src/materials/EffectiveStressInterfaceMaterial.C

```diff
@@ -0,0 +1,93 @@
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "EffectiveStressInterfaceMaterial.h"
+#include "EffectiveStressTools.h"
+#include "InterfaceValueTools.h"
+
+registerMooseObject("DeerApp", EffectiveStressInterfaceMaterial);
+
+template <> InputParameters validParams<EffectiveStressInterfaceMaterial>() {
+  InputParameters params = validParams<InterfaceMaterial>();
+  params.addClassDescription("Compute an effective stress across an interface");
+  params.addRequiredParam<MooseEnum>("effective_stress_type",
+                                     EffectiveStressTools::scalarOptions(),
+                                     "Type of effective stress to be computed");
+  params.addParam<MooseEnum>("interface_value_type",
+                             InterfaceValueTools::InterfaceAverageOptions(),
+                             "Type of scalar output");
+  params.addParam<std::vector<Real>>("params_vector",
+                                     "vector of effective stress parameters");
+  params.addParam<Point>("point1", Point(0, 0, 0),
+                         "Start point for axis used to calculate some "
+                         "cylindrical material tensor quantities");
+  params.addRequiredParam<MaterialPropertyName>(
+      "effective_stress_mp_name", "the name of the calcualte effective stress");
+  params.addParam<Point>(
+      "point2", Point(0, 1, 0),
+      "End point for axis used to calculate some material tensor quantities");
+  params.addParam<Point>("direction", Point(0, 0, 1), "Direction vector");
+  params.addParam<bool>(
+      "stateful", false,
+      "a boolean saying if we want a stateful material property or not");
+  return params;
+}
+
+EffectiveStressInterfaceMaterial::EffectiveStressInterfaceMaterial(
+    const InputParameters &parameters)
+    : InterfaceMaterial(parameters),
+      _stress_master(getMaterialPropertyByName<RankTwoTensor>("stress")),
+      _stress_slave(getNeighborMaterialPropertyByName<RankTwoTensor>("stress")),
+      _effective_stress(declareProperty<Real>(
+          getParam<MaterialPropertyName>("effective_stress_mp_name"))),
+      _effective_stress_type(getParam<MooseEnum>("effective_stress_type")),
+      _interface_value_type(parameters.get<MooseEnum>("interface_value_type")),
+      _params_vector(getParam<std::vector<Real>>("params_vector")),
+      _point1(parameters.get<Point>("point1")),
+      _point2(parameters.get<Point>("point2")),
+      _input_direction(parameters.get<Point>("direction") /
+                       parameters.get<Point>("direction").norm()),
+      _stateful(getParam<bool>("stateful")) {
+  if (_effective_stress_type == 2 && _params_vector.size() != 1)
+    mooseError("the huddleston effective stress requires the parameters b to "
```

Comment:
The usual comments on capitalization and spelling. (and continued below, i.e. hayhurs = Hayhurst).